x-pack/plugins/lens/public/indexpattern_datasource/indexpattern.tsx

```diff
@@ -383,33 +383,34 @@ export function getIndexPatternDatasource({
         return [
           ...layerErrors,
           ...realIndex.map(([filteredIndex, layerIndex]) => {
-            const fieldsWithBrokenReferences: string[] = invalidFieldsPerLayer[filteredIndex].map(
-              (columnId) => {
-                const column = invalidLayers[filteredIndex].columns[
-                  columnId
-                ] as FieldBasedIndexPatternColumn;
-                return column.sourceField;
-              }
-            );
+            const columnLabelsWithBrokenReferences: string[] = invalidColumnsForLayer[
+              filteredIndex
+            ].map((columnId) => {
+              const column = invalidLayers[filteredIndex].columns[
+                columnId
+              ] as FieldBasedIndexPatternColumn;
+              return column.label;
+            });
 
             if (originalLayersList.length === 1) {
               return {
                 shortMessage: i18n.translate(
                   'xpack.lens.indexPattern.dataReferenceFailureShortSingleLayer',
                   {
-                    defaultMessage: 'Invalid {fields, plural, one {reference} other {references}}.',
+                    defaultMessage:
+                      'Invalid {columns, plural, one {reference} other {references}}.',
                     values: {
-                      fields: fieldsWithBrokenReferences.length,
+                      columns: columnLabelsWithBrokenReferences.length,
                     },
                   }
                 ),
                 longMessage: i18n.translate(
                   'xpack.lens.indexPattern.dataReferenceFailureLongSingleLayer',
                   {
-                    defaultMessage: `{fieldsLength, plural, one {Field} other {Fields}} "{fields}" {fieldsLength, plural, one {has an} other {have}} invalid reference.`,
+                    defaultMessage: `"{columns}" {columnsLength, plural, one {has an} other {have}} invalid reference.`,
```

Comment:
Will users understand what an "invalid reference" is? I thought we discussed giving this a better label, but I can't find the discussion any more. I'm mostly asking because you've already made changes here- would be nice to resolve this naming issue.

Reply:
I'll create a doc with all the copy we have right now and we don't feel very good about.